### ISSUE TITLE
NewUser (LDAP) see all Passwords Version 2.1.27.33 - knows issus?

### ISSUE THREAD

**Issue by andreas-tr** [nilsteampassnet/TeamPass#2522]
I think it is an knows issus from the last years. #2101 
But when comes a result. So i think teampass is death?

### Steps to reproduce
1. Teampass is configured for use with ldap
2. create some users with private folders und some group folders

### Expected behaviour
When i login with new user credentials from the ldap (user doesn't exist at teampass) i can see and edit private folders from other user!

### Actual behaviour
Login with a new user (without account in teampass) shoud see only his personal folder and no other folders

### Server configuration
**Operating system**: Linux wbpweb01 4.4.0-141-generic #167-Ubuntu SMP Wed Dec 5 10:40:15 UTC 2018 x86_64

**Web server:** Apache/2.4.18 (Ubuntu)

**Database:** 5.7.24-0ubuntu0.16.04.1

**PHP version:** 7.0.32-0ubuntu0.16.04.1

**Teampass version:** 2.1.27.33

**Teampass configuration file:**
```
'max_latest_items' => '10',
'enable_favourites' => '1',
'show_last_items' => '1',
'enable_pf_feature' => '1',
'log_connections' => '1',
'log_accessed' => '1',
'time_format' => 'H:i:s',
'date_format' => 'd/m/Y',
'duplicate_folder' => '0',
'item_duplicate_in_same_folder' => '0',
'duplicate_item' => '0',
'number_of_used_pw' => '10',
'manager_edit' => '1',
'cpassman_dir' => '/var/www/teampass',
'cpassman_url' => 'http://<anonym_url>
'favicon' => 'http://<anonym_url>/favicon.ico',
'path_to_upload_folder' => '/var/www/teampass/upload',
'url_to_upload_folder' => 'http://<anonym_url>/upload',
'path_to_files_folder' => '/var/www/teampass/files',
'url_to_files_folder' => 'http://<anonym_url>/files',
'activate_expiration' => '0',
'pw_life_duration' => '0',
'maintenance_mode' => '0',
'enable_sts' => '0',
'encryptClientServer' => '1',
'cpassman_version' => '2.1.27',
'ldap_mode' => '1',
'ldap_type' => 'windows',
'ldap_suffix' => '@XXXX.local',
'ldap_domain_dn' => 'DC=XXXX,DC=local',
'ldap_domain_controler' => 'DC1.XXXX.local,DC2.XXXX.local',
'ldap_user_attribute' => 'sAMAccountName',
'ldap_ssl' => '0',
'ldap_tls' => '0',
'ldap_elusers' => '0',
'ldap_search_base' => 'OU=folder,DC=XXXX,DC=local',
'ldap_port' => '389',
'richtext' => '0',
'allow_print' => '0',
'roles_allowed_to_print' => '0',
'show_description' => '1',
'anyone_can_modify' => '1',
'anyone_can_modify_bydefault' => '1',
'nb_bad_authentication' => '5',
'utf8_enabled' => '1',
'restricted_to' => '0',
'restricted_to_roles' => '0',
'enable_send_email_on_user_login' => '0',
'enable_user_can_create_folders' => '1',
'insert_manual_entry_item_history' => '0',
'enable_kb' => '0',
'enable_email_notification_on_item_shown' => '0',
'enable_email_notification_on_user_pw_change' => '0',
'custom_logo' => 'http://<anonym_url>/logo.png',
'custom_login_text' => '',
'default_language' => 'german',
'send_stats' => '0',
'send_statistics_items' => 'stat_mysqlversion;stat_phpversion;',
'send_stats_time' => '1545055932',
'get_tp_info' => '0',
'send_mail_on_user_login' => '0',
'sending_emails' => '0',
'nb_items_by_query' => 'auto',
'enable_delete_after_consultation' => '0',
'enable_personal_saltkey_cookie' => '0',
'personal_saltkey_cookie_duration' => '60',
'email_smtp_server' => 'int-mail.XXXX.local',
'email_smtp_auth' => '',
'email_auth_username' => '',
'email_auth_pwd' => '<removed>'
'email_port' => '',
'email_security' => '',
'email_server_url' => '',
'email_from' => 'teampass@domain.de',
'email_from_name' => 'Teampass',
'pwd_maximum_length' => '40',
'google_authentication' => '0',
'delay_item_edition' => '0',
'allow_import' => '0',
'proxy_ip' => 'firewall.XXXX.local',
'proxy_port' => '8080',
'upload_maxfilesize' => '10mb',
'upload_docext' => 'docx,dotx,xlsx,xltx,rtf,csv,txt,pdf,pptx,pot,dotx,xltx',
'upload_imagesext' => 'jpg,jpeg,gif,png',
'upload_pkgext' => '7z,rar,tar,zip',
'upload_otherext' => 'sql,xml',
'upload_imageresize_options' => '1',
'upload_imageresize_width' => '800',
'upload_imageresize_height' => '600',
'upload_imageresize_quality' => '90',
'use_md5_password_as_salt' => '0',
'ga_website_name' => 'TeamPass for ChangeMe',
'api' => '0',
'subfolder_rights_as_parent' => '1',
'show_only_accessible_folders' => '1',
'enable_suggestion' => '0',
'otv_expiration_period' => '7',
'default_session_expiration_time' => '120',
'duo' => '0',
'enable_server_password_change' => '0',
'ldap_object_class' => 'organizationalPerson',
'bck_script_path' => '/var/www/teampass/backups',
'bck_script_filename' => 'bck_teampass',
'syslog_enable' => '0',
'syslog_host' => 'localhost',
'syslog_port' => '514',
'manager_move_item' => '0',
'create_item_without_password' => '1',
'otv_is_enabled' => '0',
'agses_authentication_enabled' => '0',
'item_extra_fields' => '0',
'saltkey_ante_2127' => 'none',
'migration_to_2127' => 'done',
'files_with_defuse' => 'done',
'timezone' => 'Europe/Berlin',
'enable_attachment_encryption' => '1',
'personal_saltkey_security_level' => '50',
'ldap_new_user_is_administrated_by' => '0',
'disable_show_forgot_pwd_link' => '1',
'offline_key_level' => '60',
'enable_http_request_login' => '0',
'ldap_and_local_authentication' => '0',
'secure_display_image' => '1',
'upload_zero_byte_file' => '0',
'upload_all_extensions_file' => '0',
'bck_script_passkey' => '<removed>'
'admin_2fa_required' => '1',
'ldap_bind_dn' => 'CN=usersearch,OU=folder,DC=XXXX,DC=local',
'ldap_bind_passwd' => '<removed>'
'use_http_request_login' => '1',
'settings_offline_mode' => '0',
'ldap_new_user_role' => '4',

```

**Updated from an older Teampass or fresh install:**
Fresh Installation

### Client configuration

**Browser:** Firefox - 64.0

**Operating system:** Windows - 64bits

#### Web server error log
```
Undefined index: path - /var/www/teampass/sources/main.queries.php (1353)
```

#### Teampass 10 last system errors
```

```

#### Log from the web-browser developer console (CTRL + SHIFT + i)
```
Insert the log here and especially the answer of the query that failed.
```

**Comment by nilsteampassnet** [comment]
I don't understand as this version includes the fix for #2499.
Performing new tests just now, I cannot reproduce it anymore.
Perhaps is there something wrong in this package (bad merge) ... i need to check.

**Comment by andreas-tr** [comment]
Did you know when you release a bugfix?

**Comment by lduriez** [comment]
I have the exact same problem with this server and client configuration:

### Server configuration
**Operating system**:
Debian jessie 8.11

**Web server:**
Apache 2.4.10

**Database:**
Mysql 5.5.62

**PHP version:**
5.6.40

**Teampass version:** 
2.1.27.33

### Client configuration
**Browser:**
Firefox 65.0 (64bits)

**Operating system:**
Ubuntu 18.04.2 LTS bionic

**Comment by lduriez** [comment]
I upgrade to 2.1.27.34 version, the problem still occur.
I see all personal folder but not identified (they are like 11110 11111 11112 ...), I see the information except the password on the folder where for the one I got salt-key in cookies.
I can change or create new registration because the salt-key is require.

**Comment by andreas-tr** [comment]
I upgrade to Version 2.1.27.35 , the problem still occur!

**Comment by nilsteampassnet** [comment]
I investigated this point again.
I did found one bad init in source code.
I have run successfully some tests with my ldap server.
I would be interest in any other experimentation.

The change is quite simple.

* open file `/sources/main.functions.php`
* Replace this block
```
            if (empty($persoFld['id']) === false && DB::count() > 0) {
                if (in_array($persoFld['id'], $listAllowedFolders) === false) {
                    array_push($_SESSION['personal_folders'], $persoFld['id']);
                    array_push($listAllowedFolders, $persoFld['id']);
                    array_push($_SESSION['personal_visible_groups'], $persoFld['id']);
                    
                    // get all descendants
                    $ids = $tree->getDescendants($persoFld['id'], false, false);
                    foreach ($ids as $ident) {
                        if ((int) $ident->personal_folder === 1) {
                            array_push($listAllowedFolders, $ident->id);
                            array_push($_SESSION['personal_visible_groups'], $ident->id);
                            array_push($_SESSION['personal_folders'], $ident->id);
                        }
                    }
                }
            }
```
by
```
            if (empty($persoFld['id']) === false && DB::count() > 0) {
                if (in_array($persoFld['id'], $listAllowedFolders) === false) {
                    array_push($_SESSION['personal_folders'], $persoFld['id']);
                    array_push($listAllowedFolders, $persoFld['id']);
                    array_push($_SESSION['personal_visible_groups'], $persoFld['id']);
                    
                    // get all descendants
                    $ids = $tree->getDescendants($persoFld['id'], false);
                    foreach ($ids as $ident) {
                        if ((int) $ident->personal_folder === 1) {
                            array_push($listAllowedFolders, $ident->id);
                            array_push($_SESSION['personal_visible_groups'], $ident->id);
                            array_push($_SESSION['personal_folders'], $ident->id);
                        }
                    }
                }
            }
```

**Comment by andreas-tr** [comment]
Hey i have test it again. But it didn't work.
![grafik](https://user-images.githubusercontent.com/46756391/54594006-01c3b700-4a30-11e9-82d1-fa95fcb09e6c.png)

**Comment by lduriez** [comment]
It tried it as well, and the same problem hapen.
![image](https://user-images.githubusercontent.com/36005270/54599806-b31c1a00-4a3b-11e9-83a9-6affb9963481.png)

**Comment by lo-man** [comment]
Same here. I can see and open the privat folders from all other users. After a browser refresh (F5) the other folders are gone. This behavior is since month. 
TeamPass 2.1.27.35

**Comment by andreas-tr** [comment]
Now 6 Month after open the bug case the problem still occur!
At the moment the software is unsuable! So what now?

**Comment by lduriez** [comment]
I tried with the docker version and the problem same problem occured

**Comment by amdnoc** [comment]
For me this worked:



```
 if (empty($persoFld['id']) === false && DB::count() > 0) {
                if (in_array($persoFld['id'], $listAllowedFolders) === false) {
                    array_push($_SESSION['personal_folders'], $persoFld['id']);
                    array_push($listAllowedFolders, $persoFld['id']);
                    array_push($_SESSION['personal_visible_groups'], $persoFld['id']);
                    
                    // get all descendants
                    // FIX get only children
                    $ids = $tree->getDescendants($persoFld['id'], false,true);
                    foreach ($ids as $ident) {
                        if ((int) $ident->personal_folder === 1) {
                            array_push($listAllowedFolders, $ident->id);
                            array_push($_SESSION['personal_visible_groups'], $ident->id);
                            array_push($_SESSION['personal_folders'], $ident->id);
                        }
                    }
                }
            }

```

**Comment by lduriez** [comment]
I tested what amdnoc post and it work for me, thanks.
I just have now a personnal folder with a number id, but it's peanut and much better than before.

![image](https://user-images.githubusercontent.com/36005270/62762263-faf21b00-ba88-11e9-9d21-0f5725c374c6.png)

**Comment by lduriez** [comment]
Pull Request -> #2663

**Comment by ewheelerinc** [comment]
Just wanted to mention that we had a similar issue.  We configured the roles to restrict a user to only that role-set, but they still saw all folders.

I logged into admin, edited the user, clicked "allowed folders" and clicked "uncheck all".  Now the user can only see folders in their role. Not sure if this was just an issue on my end, but check the user's "allowed folders" list if you hit a bug like this.

-Eric
